tests/page_test.go

```diff
@@ -1873,3 +1882,187 @@ func TestPageGetAttributeEmpty(t *testing.T) {
 	require.True(t, ok)
 	assert.Equal(t, "", got)
 }
+
+func TestPageOnMetric(t *testing.T) {
+	t.Parallel()
+
+	// This page will perform many pings with a changing h query parameter.
+	// This URL should be grouped according to how page.on('metric') is used.
+	tb := newTestBrowser(t, withHTTPServer())
+	tb.withHandler("/home", func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, `
+		<html>
+			<head></head>
+			<body>
+				<script type="module">
+					await ping();
+					async function ping() {
+						await fetch('/ping?h=2kq2lo6n06');
+						await fetch('/ping?h=ej0ypprcjk');
+					}
+				</script>
+			</body>
+		</html>`)
+		require.NoError(t, err)
+	})
+	tb.withHandler("/ping", func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, `pong`)
+		require.NoError(t, err)
+	})
+
+	ignoreURLs := map[string]any{
+		tb.url("/home"):        nil,
+		tb.url("/favicon.ico"): nil,
+	}
+
+	tests := []struct {
+		name string
+		fun  string
+		want string
+	}{
+		{
+			// Just a single page.on.
+			name: "single_page.on",
+			fun: `page.on('metric', (metric) => {
+				metric.Tag({
+				  urls: [
+						{urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-1'},
+					]
+				});
+			});`,
+			want: "ping-1",
+		},
+		{
+			// A single page.on but with multiple calls to Tag.
+			name: "multi_tag",
+			fun: `page.on('metric', (metric) => {
+				metric.Tag({
+				  urls: [
+						{urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-1'},
+					]
+				});
+				metric.Tag({
+					urls: [
+						  {urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-2'},
+					  ]
+				  });
+			});`,
+			want: "ping-2",
+		},
+		{
+			// Two page.on and in one of them multiple calls to Tag.
+			name: "multi_tag_page.on",
+			fun: `page.on('metric', (metric) => {
+				metric.Tag({
+				  urls: [
+						{urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-1'},
+					]
+				});
+				metric.Tag({
+					urls: [
+						  {urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-2'},
+					  ]
+				  });
+			});
+			page.on('metric', (metric) => {
+				metric.Tag({
+				  urls: [
+						{urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-3'},
+					]
+				});
+			});`,
+			want: "ping-3",
+		},
+		{
+			// A single page.on but within it another page.on.
+			name: "multi_page.on_call",
+			fun: `page.on('metric', (metric) => {
+				metric.Tag({
+				  urls: [
+						{urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-1'},
+					]
+				});
+				page.on('metric', (metric) => {
+					metric.Tag({
+						urls: [
+							{urlRegEx: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, tagName:'ping-4'},
+						]
+					});
+				});
+			});`,
+			want: "ping-4",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Nice tests, but do we forget to call `t.Parallel`?

Reply:
Good spot, done in bcc15bc7cd5b6e10835688d650a6a5613a662b01